.gitignore

```diff
@@ -6,3 +6,7 @@
 
 # Ignore built documentation
 gh-build
+
+# Ignore common build directory names
+bld
+build
```

Comment:
This seems wise, but I notice that this PR has files in a directory named `build`

Reply:
Good catch. Pretty sure `.gitignore` entires are recursive, so any changes to any files in directories called `build` (of which there are several) will not be picked up if this line remains.

Reply:
Actually - if they are already tracked, then git will still pick up changes.  If new files are added in such a directory, then they will not be noticed, and a developer will need to be carefully to manually add them

I propose that we stop using `build` as the directory name for source

Reply:
I agree and created #644 so this can be addressed in a future PR.